pacman/operations/routing_info_allocator_algorithms/global_zoned_routing_info_allocator.py

```diff
@@ -0,0 +1,184 @@
+# Copyright (c) 2019 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import division
+import math
+from spinn_utilities.progress_bar import ProgressBar
+from pacman.model.routing_info import (
+    RoutingInfo, PartitionRoutingInfo, BaseKeyAndMask)
+from pacman.utilities.utility_calls import (
+    check_algorithm_can_support_constraints)
+from pacman.exceptions import PacmanRouteInfoAllocationException
+from pacman.model.constraints.key_allocator_constraints import (
+    AbstractKeyAllocatorConstraint, ContiguousKeyRangeContraint)
+from pacman.model.graphs.common import EdgeTrafficType
+
+KEY_SIZE = 32
+
+
+class GlobalZonedRoutingInfoAllocator(object):
+    """ A routing key allocator that uses fixed zones that are the same for\
+        all vertices
+
+    .. note::
+        No constraints are supported, and that the number of keys\
+        required by each edge must be 2048 or less, and that all edges coming\
+        out of a vertex will be given the same key/mask assignment.
+    """
+
+    __slots__ = [
+        # Passed in parameters
+        "_application_graph",
+        "_graph_mapper",
+        "_machine_graph",
+        "_placements",
+        "_n_keys_map",
+        "_n_bits_partition",
+        "_n_bits_machine",
+        "_n_bits_atoms"
+    ]
+    # pylint: disable=attribute-defined-outside-init
+
+    def __call__(self, application_graph, graph_mapper, machine_graph,
```

Comment:
still nope on the graph mapper

Reply:
(cwl) its not even doced in the call docs. so least you dont need to fix that. but still. bad on you for not docing the dead param. lol